# tfs-ruby-3-1-x-gem-prelude-tebako-runtime.patch
#
# Target:  gem_prelude.rb (pass1a (all platforms))
# Applies: ruby 3.1.x (3.1.6)
# Source:  tebako gem lib/tebako/packager/pass1a_patch.rb Pass1APatch::GEM_PRELUDE_RB_PATCH
#
# Require tebako-runtime at interpreter startup (hooked next to the
# DidYouMean definition in gem_prelude); warn if it cannot be loaded.
# X-Redesign: modern-api
diff --git a/gem_prelude.rb b/gem_prelude.rb
index 94ada31..40f76b4 100644
--- a/gem_prelude.rb
+++ b/gem_prelude.rb
@@ -17,3 +17,12 @@ begin
 rescue LoadError
   warn "`did_you_mean' was not loaded."
 end if defined?(DidYouMean)
+
+# -- Start of tebako patch --
+begin
+  require 'tebako-runtime'
+rescue LoadError
+  warn "'tebako-runtime' was not loaded."
+end
+# -- End of tebako patch --
+
